lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -34,7 +34,10 @@ sub register ($self, $app, $config) {
         format_time_duration => sub {
             my ($c, $timedate) = @_;
             return unless $timedate;
-            if ($timedate->hours() > 0) {
+            if ($timedate->days() > 0) {
+                sprintf("%d days %02d:%02d hours", $timedate->days(), $timedate->hours(), $timedate->minutes());
+            }
+            elsif ($timedate->hours() > 0) {
                 sprintf("%02d:%02d hours", $timedate->hours(), $timedate->minutes());
             }
```

Comment:
hm, maybe we don't need to repeat the formating for hours. How about doing the hours block in both cases and in case of days add that as prefix. Something along the lines of

```suggestion
            if ($timedate->… > 0) {
                my $out = sprintf("%02d:%02d hours", $timedate->hours(), $timedate->minutes());
                $out = $timedate->days() . ' days ' . $out if $timedate->days();
                return $out;
            }
```

hm, ok. maybe not really better ;) Maybe you have a better idea.

Reply:
Then the condition needs to check for `days > 0 || hours > 0`.

Reply:
I do not feel it is really better to make use of a prefix. No code reduction and harder to read.